Slim/Schema/Genre.pm

```diff
@@ -35,6 +39,34 @@ use Slim::Utils::Log;
 	$class->resultset_class('Slim::Schema::ResultSet::Genre');
 }
 
+sub loadMyClassicalGenreMap {
+	my $prefs = preferences('server');
+	%myClassicalGenreMap = map {$_ => 1} split(/,\s*/, uc($prefs->get('myClassicalGenres')));
+	# also load genre ids from database
+	my @genreNames = keys %myClassicalGenreMap;
+	my $dbh = Slim::Schema->dbh;
+	my $sql = 'SELECT GROUP_CONCAT(id) FROM genres WHERE UPPER(name) IN (' . join(', ', map {'?'} @genreNames) . ')';
+	my $sth = $dbh->prepare_cached($sql);
+	$sth->execute(@genreNames);
+	($myClassicalGenreIds) = $sth->fetchrow_array;
+	$sth->finish;
+}
+
+sub isMyClassicalGenre {
+	loadMyClassicalGenreMap() if !%myClassicalGenreMap;
```

Comment:
Like above: we have to distinguish between an non-initialised list, and an empty list of genres to prevent unnecessary parsing.